modules/dxp/apps/portal-search-similar-results/portal-search-similar-results-web/src/main/java/com/liferay/portal/search/similar/results/web/internal/portlet/shared/search/SimilarResultsPortletSharedSearchContributor.java

```diff
@@ -168,16 +173,12 @@ private void _filterByEntryClassName(
 			return;
 		}
 
-		Optional<String> classNameOptional = criteria.getTypeOptional();
+		String className = criteria.getType();
 
-		classNameOptional.ifPresent(
-			className -> {
-				if (!Validator.isBlank(className)) {
-					searchRequestBuilder.addComplexQueryPart(
-						_getComplexQueryPart(
-							_getEntryClassNameQuery(className)));
-				}
-			});
+		if ((className != null) && !Validator.isBlank(className)) {
+			searchRequestBuilder.addComplexQueryPart(
+				_getComplexQueryPart(_getEntryClassNameQuery(className)));
+		}
 	}
 
```

Comment:
@slnn By the way, I can see from the git history that you modified `_populate` method in this class, but finally it shows unmodified. Please avoid such modification in git history. I guess it's because you modified `SearchStringUtil` in multiple steps, and applied usage in each step. Actually, no need to do that if the modification is not very complex, or involving logic refactoring. Please clean up your commits to reduce such history, thanks.